### PR TITLE
feat(cms): Orders api config and edit function

### PR DESCRIPTION
**Pull request by limivann**
- Integrated functionality to fetch and display order data via the Orders API.
- Added features to edit and delete individual orders directly from the orders page.

### CODE REVIEW

**Review by Iyzyman — COMMENTED** [review]
LGTM, but can I ask why you used Class for Order instead of Type?

**Review by Iyzyman — DISMISSED** [review]

**Comment by limivann** [comment]
> LGTM, but can I ask why you used Class for Order instead of Type?

We used Class instead of Type to iterate the keys of Order

```
 for (const key of Object.keys(new Order())) {
    const renderCellComponent = RenderCellFactory.get(data[0], key);
    const renderCell: React.FC<{ children?: React.ReactNode }> =
      renderCellComponent instanceof Promise
        ? renderCellComponent
        : renderCellComponent;
```

File: apps/cms/src/@types/Order.ts [hunk 1]
```diff
@@ -4,14 +4,13 @@
...
-    public colour = ''
-
+    public id = '',
+    public transaction_id = '',
+    public transaction_time = '',
+    public payment_method = '',
+    public customerEmail = '',
+    public status = '',
+    public updatedAt = '',
+    public createdAt = ''
```
Comment (chanbakjsd) [inline apps/cms/src/@types/Order.ts:14]: Is there any reason some fields are `snake_case` and some fields are `camelCase`?

File: apps/cms/src/admin/views/MerchSales.tsx [hunk 2]
```diff
@@ -5,19 +5,28 @@ import ViewTemplate from "./ViewTemplate";
...
+  const [data, setData] = useState<Order[]>(null);
+  const history = useHistory();
+  useEffect(() => {
+    const fetchOrders = async () => { 
+      try {
+        const orders: Order[] = await OrdersApi.getOrders();
+        setData(orders);
+      } catch (error) {
+        console.error(error);
+        setData([]);
```
Comment (chanbakjsd) [inline apps/cms/src/admin/views/MerchSales.tsx:25]: Can we do better error handling than this? Maybe even just displaying an error message?

File: apps/web/features/merch/services/api.ts [hunk 3]
```diff
@@ -93,7 +93,7 @@ export class Api {
     return new Promise((res, rej) => {
       setTimeout(() => {
         res({
-          disabled: true,
+          disabled: false,
```
Comment (chanbakjsd) [inline apps/web/features/merch/services/api.ts:96]: This doesn't seem like an intended change.

**Review by chanbakjsd — CHANGES_REQUESTED** [review]
There are just a few issues to address :)

File: apps/cms/src/@types/Order.ts (hunk 1 above)
Comment (limivann) [reply apps/cms/src/@types/Order.ts:14]: No idea, it was already inconsistent prior to this PR: https://github.com/ntuscse/website/pull/153

File: apps/web/features/merch/services/api.ts (hunk 3 above)
Comment (limivann) [reply apps/web/features/merch/services/api.ts:96]: Noted, will revert

File: apps/cms/src/@types/Order.ts (hunk 1 above)
Comment (chanbakjsd) [reply apps/cms/src/@types/Order.ts:14]: In that case, can you help updating it to be consistent? Thank you!

**Comment by limivann** [comment]
Hi @chanbakjsd , I have made the following changes:
- added toast messages to display error messages when triggered, using react-toastify because payloadcms is also using it internally
- fixed inconsistency issue of snake case and camel case, also changed implementation of prettifyKeys to convert camel to human readable titles
- reverted unnecessary changes

CC: @Iyzyman

**Review by chanbakjsd — APPROVED** [review]

**Review by Iyzyman — APPROVED** [review]

**Pull request merged**